### DISCUSSION TITLE
Error when trying to install scikit-learn into "my-rdkit-env" that contains rdkit package

### DISCUSSION THREAD

**Discussion by anqima-ai** [rdkit/rdkit#5242] [Q&A] [answered]
Hi,
I am new to rdkit.
Following Richard L. Apodaca's instructions on depth-first.com. I successfully installed rdkit and jupyter notebook.

When I am trying to others' code that needs scikit-learn, I realized the default rdkit package seems not to have scikit-learn.
I tried to use conda to install scikit-learn in the same env "my-rdkit-env". It started to give error like:

"Collecting package metadata (current_repodata.json): done
Solving environment: failed with initial frozen solve. Retrying with flexible solve.
Solving environment: failed with repodata from current_repodata.json, will retry with next repodata source."

I created another env like "testenv" and have no issue to install scikit-learn using conda.

After comparing versions, in the scikit-learn package the version of python is 3.9.12. However, in my rdkit package, the version of python is 3.10.4.
I am wondering if compatibility is one of the reasons that I can't successfully install scikit-learn with rdkit in the same environment.

Does anyone have some quick solution to this issue?  Thanks in advance!

**Comment by DavidACosgrove** [comment, marked as answer]
Hi,

I have just successfully installed both packages with:

conda create --name RDKit.test -c conda-forge rdkit python=3.9

conda activate RDKit.test

conda install scikit-learn


I didn't attempt to install the default python, based on your experience,
this was just to show how to use conda to select a specific version of
python.


Dave



On Thu, Apr 28, 2022 at 7:03 AM anqima-ai ***@***.***> wrote:

> Hi,
> I am new to rdkit.
> Following Richard L. Apodaca's instructions on depth-first.com. I
> successfully installed rdkit and jupyter notebook.
>
> When I am trying to others' code that needs scikit-learn, I realized the
> default rdkit package seems not to have scikit-learn.
> I tried to use conda to install scikit-learn in the same env
> "my-rdkit-env". It started to give error like:
>
> "Collecting package metadata (current_repodata.json): done
> Solving environment: failed with initial frozen solve. Retrying with
> flexible solve.
> Solving environment: failed with repodata from current_repodata.json, will
> retry with next repodata source."
>
> I created another env like "testenv" and have no issue to install
> scikit-learn using conda.
>
> After comparing versions, in the scikit-learn package the version of
> python is 3.9.12. However, in my rdkit package, the version of python is
> 3.10.4.
> I am wondering if compatibility is one of the reasons that I can't
> successfully install scikit-learn with rdkit in the same environment.
>
> Does anyone have some quick solution to this issue? Thanks in advance!
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/rdkit/rdkit/discussions/5242>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACGF2FVD7RCLGN5EBKZBI4LVHIS4TANCNFSM5URIFW5Q>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>


-- 
David Cosgrove
Freelance computational chemistry and chemoinformatics developer
http://cozchemix.co.uk

**Reply by anqima-ai** [reply]
Hi Dave,

Yes. It worked for me now this morning following your suggestion.

As I guessed and you said, most likely it's due to the conflict between rdkit and scikit-learn that need different version of python.

I created another env as you suggested "rdkit-sklearn-env" to host both rdkit and scikit-learn, and will keep my old "my-rdkit-env" for rdkit using python 3.10 without having scikit-learn.

Thank you so much for your life-saving answer and running the installation for me!

Best,

Anqi
